adagrams/game.py

```diff
@@ -1,11 +1,154 @@
+import random
+LETTER_POOL = [{
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}]
 def draw_letters():
-    pass
+    letters = []
+    remaining_letters_pool = {}
+    
+    if len(letters) < 11:
+    #getting the remaining amount of letters in the letter pool
+        for dict in LETTER_POOL:
+            for key, value in dict.items():
+                if key not in remaining_letters_pool and value > 0:
+                 remaining_letters_pool[key] = value
+    #adding random letter to the length of the letter list if it's less than ten
+    while len(letters) < 10:
+        random_letter = random.choice(list(remaining_letters_pool))
+    ##Removing the used letter from the remaining letter pool  
+        for letter in list(remaining_letters_pool):
+            if remaining_letters_pool[letter] == 0:
+                remaining_letters_pool.pop(letter)
+            elif random_letter == letter:
+                remaining_letters_pool[letter]-= 1
+                letters.append(random_letter)            
+    return letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    dict_word={}
+    word=word.upper()
+    word_list=[]
+    values_dict=[]
+    #Creating a list of each letter that is a string
+    for letter in word:
+        word_list.append(letter)
+    #Assigning a boolean to each letter if it appears in the list
+    for letter in word_list:
+        if letter in letter_bank:
+            freq_letter=word_list.count(letter)
+            freq_letter_bank=letter_bank.count(letter)
+            if freq_letter == freq_letter_bank:
+                dict_word[letter]=True
+            elif freq_letter>freq_letter_bank:
+                dict_word[letter]=False
+            elif freq_letter<freq_letter_bank:
+                dict_word[letter]=True
+        else:
+            dict_word[letter]=False
+    #Adding all of the booleans values for the given word to a list
+    for key,value in dict_word.items():
+        values_dict.append(value)
+    if False in values_dict:
+    #if there are any False values in the dict then return False
+        return False
+    else:
+        return True
+
```

Comment:
I see what y'all were trying to do here, and the code works and shows a good understanding of lists and dictionaries, but there's a simpler way to approach this problem and it starts with creating a deep copy of the letter bank. That would allow us to have a copy of the letter bank that we can modify without modifying the original. From there, you could simply loop through each letter in the word and check to see if it is in the copied letter_bank. If it is in there, remove the letter from the letter bank and move to the next one (That'll take care of duplicate letters). If it isn't there, we can immediately return false. That would look something like this:

     for letter in word:
          if letter in letter_bank_copy:
               letter_bank_copy.remove(letter)
         else: 
               return False
     return True